examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect.ino

```diff
@@ -0,0 +1,591 @@
+    LogError(logString + result);
+    exit(result);
+  }
+
+  // Sign and encode the signature (b64 encoded, HMAC-SHA256 signing).
+  // Uses the decoded device key.
+  generateSASBase64EncodedSignedSignature(
+      az_span_ptr(signatureAzSpan), az_span_size(signatureAzSpan),
+      encodedSignedSignature, sizeof(encodedSignedSignature), &encodedSignedSignatureLength);
+
+  // Get the MQTT password (SAS Token) from the base64 encoded, HMAC signed bytes.
+  az_span encodedSignedSignatureAzSpan = az_span_create(encodedSignedSignature, 
+                                                        encodedSignedSignatureLength);
+  result = az_iot_hub_client_sas_get_password(
+      &azIoTHubClient, sasTokenDuration, encodedSignedSignatureAzSpan, AZ_SPAN_EMPTY,
+      mqttPassword, sizeof(mqttPassword), NULL);
+  if (az_result_failed(result)) 
+  {
+    logString = "Could not get the MQTT password. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+}
+
+/*
+ * generateSASBase64EncodedSignedSignature:
+ * Sign and encode a signature. It is signed using the provided device key.
+ * The process is:
+ *    1. Decode the encoded device key.
+ *    2. Sign the signature with the decoded device key.
+ *    3. Encode the signed signature.
+ */
+static void generateSASBase64EncodedSignedSignature(
+    uint8_t const* sasSignature, size_t const sasSignatureSize,
+    uint8_t* encodedSignedSignature, size_t encodedSignedSignatureSize,
+    size_t* encodedSignedSignatureLength) 
+{
+  int result;
+  unsigned char sasDecodedKey[BUFFER_LENGTH_SAS] = {0};
+  size_t sasDecodedKeyLength = 0;
+  unsigned char sasHMAC256SignedSignature[BUFFER_LENGTH_SAS] = {0};
+  mbedtls_md_context_t ctx;
+  mbedtls_md_type_t md_type = MBEDTLS_MD_SHA256;
+
+  // Decode the SAS base64 encoded device key to use for HMAC signing.
+  result = mbedtls_base64_decode(
+      sasDecodedKey, sizeof(sasDecodedKey), &sasDecodedKeyLength,
+      (const unsigned char*)IOT_CONFIG_DEVICE_KEY, sizeof(IOT_CONFIG_DEVICE_KEY) - 1);
+  if (result != 0) 
+  {
+    logString = "mbedtls_base64_decode failed. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+
+  // HMAC-SHA256 sign the signature with the decoded device key.
+  mbedtls_md_init(&ctx);
+  mbedtls_md_setup(&ctx, mbedtls_md_info_from_type(md_type), 1);
+  mbedtls_md_hmac_starts(&ctx, sasDecodedKey, sasDecodedKeyLength);
+  mbedtls_md_hmac_update(&ctx, sasSignature, sasSignatureSize);
+  mbedtls_md_hmac_finish(&ctx, sasHMAC256SignedSignature);
+  mbedtls_md_free(&ctx);
+
+  // Base64 encode the result of the HMAC signing.
+  result = mbedtls_base64_encode(
+      encodedSignedSignature, encodedSignedSignatureSize, encodedSignedSignatureLength,
+      sasHMAC256SignedSignature, sizeof(sasHMAC256SignedSignature));
+  if (result != 0) 
+  {
+    logString = "mbedtls_base64_encode failed. Return code: ";
+    LogError(logString + result);
+    exit(result);

```

Comment:
Same comment for exit. The correct implementation should be getting into an infinite loop, which is then debuggable in platforms that so support it.